internals/overlord/state/identities.go

```diff
@@ -59,6 +62,10 @@ type BasicIdentity struct {
 	Password string
 }
 
+type CertificateIdentity struct {
+	Certificate *x509.Certificate
```

Comment:
Why not make this ```x509```. This would allow a certificate identity to support other formats in the future ? This resolves the ```x.Certificate.Certificate``` problem ?